project-base/storefront/components/Layout/Header/MobileMenu/MobileMenu.tsx

```diff
@@ -0,0 +1,74 @@
+import { DropdownMenuListItem } from './MobileMenuListItem';
+import { SubMenu } from './MobileMenuSubItems';
+import { mapNavigationMenuItems } from './helpers';
+import { ArrowIcon, CloseIcon } from 'components/Basic/Icon/IconsSvg';
+import { NavigationQueryApi } from 'graphql/generated';
+import useTranslation from 'next-translate/useTranslation';
+import { useState } from 'react';
+
+export type MenuItem = {
+    name: string;
+    link: string;
+    parentItem?: string;
+    children?: MenuItem[];
+};
+
+type MobileMenuProps = {
+    navigationItems: NavigationQueryApi['navigation'];
+    onMenuToggleHandler: () => void;
+};
+
+export const MobileMenu: FC<MobileMenuProps> = ({ navigationItems, onMenuToggleHandler }) => {
+    const { t } = useTranslation();
+    const [historyMenuGroups, setHistoryMenuGroups] = useState<MenuItem[][] | undefined>();
+    const [currentMenuItems, setCurrentMenuItems] = useState<MenuItem[]>(mapNavigationMenuItems(navigationItems));
+
+    const currentGroupTitle = currentMenuItems[0].parentItem;
+
+    return (
+        <>
+            <div className="py-3 flex">
+                {!!historyMenuGroups?.length && (
+                    <button
+                        className="cursor-pointer flex w-9 justify-start gap-2 items-center text-sm uppercase text-dark"
+                        title={t('Back')}
+                        onClick={() => {
+                            setCurrentMenuItems([...historyMenuGroups].pop()!);
+                            setHistoryMenuGroups(historyMenuGroups.slice(0, -1));
```

Comment:
I am just thinking if there isn't a nicer way to write this...

Maybe the second row could use the current value provided by the anonymous function? Not sure. It's just for you to consider, you can resolve as you wish.

Reply:
Take a look. I think it was little bit improved.